### ISSUE TITLE
모델 결과 문제

### ISSUE THREAD

**Issue by qudgns5129** [qudgns5129/pill_classification#6]
우선 AlexNet을 돌려보았을 때, 성능이 안나오는 것을 확인하였고,  
![image](https://user-images.githubusercontent.com/47845170/99377668-d3376700-2909-11eb-9b73-05bc1eee2be2.png)
모델은 batch size = 8, epochs = 64, optimizer = adam, loss function = categorical_crossentropy를 사용  
batch size가 클 경우 memory error가 발생함... 이 문제도 해결해야함(resourceExhaustedError)  
  
  
-> 이에 대해 데이터의 개수가 적은 것을 이유로 생각하였습니다.  
  
  
또한 혈액 대용제, 기타의말초신경용약 등의 특정 알약에 대한 데이터가 10개가 넘지 않는 것을 확인하였습니다.  
![image](https://user-images.githubusercontent.com/47845170/99377629-c3b81e00-2909-11eb-978b-89ee9ed700cc.png)
  
  
한번 이에 대한 데이터를 삭제하고 진행한 이후, 성능이 더 향상되지 않는지 확인해보겠습니다.  
  
그리고 예측 변수에 대해 다시 한 번 생각해 보았습니다. 모든 약이 원형이기 때문에 약의 색상만으로 약의 분류명을 예측하는 과제인데, 성능이 안나오는게
- 단순히 데이터 개수의 부족으로 인해 발생하는 건지
- 색상과 약의 분류명은 상관관계가 전혀 없어서 발생하는 현상인지  

에 대한 고찰을 해보았습니다. 근데 http://m.dailypharm.com/newsView.html?ID=198983 이 사이트에 의거하면 약의 색상이 약의 분류명과 관련이 있다고 합니다.

**Comment by bytecell** [comment]
모델의 input 으로 들어간 게 뭐야?
이미지만 들어간 거야?

**Comment by qudgns5129** [comment]
이미지만 들어갔습니다

**Comment by qudgns5129** [comment]
모델 구성 시 weight는 모두 초기화 한 뒤, 구조만 가져왔습니다.

**Comment by bytecell** [comment]
기존 학습된 weight 가져와서 초기화한 후,
네 데이터를 사용해서 다시 학습(fine-tuning)해보도록 해.

**Comment by bytecell** [comment]
R, G, B 3개 채널의 픽셀 값이 입력으로 들어간 거야?

**Comment by qudgns5129** [comment]
네! 이미지 shape = (18170, 100, 100, 3) 입니다. 이미지 크기를 줄이는 것도 memery error때문에 고려해봐야할 거 같습니다.

**Comment by qudgns5129** [comment]
기존 학습된 weight를 가져온 후에 초기화를 시키고, fine-tuning을 하는 것과, 구조만 가져온 후 tuning을 하는 거는 다른 방식인가요?

**Comment by bytecell** [comment]
Weight 가져와서 전이학습 하는거야.

**Comment by bytecell** [comment]
Weight 가져와야하니 이미지 크기는 바꾸지말고 배치사이즈만 조절해.

**Comment by qudgns5129** [comment]
미디어랩스 컴퓨터로 진행중인데 batch size가 지금 32,16,8일때 안돌아가고, 4일때 하나는 돌아가는데 그 뒤의 모델을 하려하면 메모리 부족 에러가 뜹니다ㅠㅠ  
모델이 원하는 데이터 shape는 (227,227,3) 크기입니다.

**Comment by qudgns5129** [comment]
- 데이터가 너무 없는 레이블 없애기
- 227,227,3 크기로 조정하기
- 배치사이즈 낮추기  

이 3가지 진행해본 다음에 모델 학습 진행해보겠습니다.

**Comment by bytecell** [comment]
서버실 컴퓨터사용중? 몇번?

**Comment by qudgns5129** [comment]
ML313 컴퓨터입니다.
